Source/Urho3D/IO/Deserializer.cpp

```diff
@@ -387,9 +390,37 @@ Variant Deserializer::ReadVariant(VariantType type)
     case VAR_DOUBLE:
         return Variant(ReadDouble());
 
-        // Deserializing custom values is not supported. Return empty
-    case VAR_CUSTOM_HEAP:
     case VAR_CUSTOM_STACK:
+    {
```

Comment:
I suggest you have single case block for both VAR_CUSTOM_STACK and VAR_CUSTOM_HEAP since they are just implementation details of Variant. If you want to have safety checks, they better be somewhere else.